rand_distr/src/dirichlet.rs

```diff
@@ -121,27 +276,44 @@ where
     Open01: Distribution<F>,
 {
     fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Vec<F> {
-        let n = self.alpha.len();
-        let mut samples = vec![F::zero(); n];
-        let mut sum = F::zero();
-
-        for (s, &a) in samples.iter_mut().zip(self.alpha.iter()) {
-            let g = Gamma::new(a, F::one()).unwrap();
-            *s = g.sample(rng);
-            sum =  sum + (*s);
+        match &self.repr {
+            DirichletRepr::FromGamma(dirichlet) => dirichlet.sample(rng),
+            DirichletRepr::FromBeta(dirichlet) => dirichlet.sample(rng),
         }
-        let invacc = F::one() / sum;
-        for s in samples.iter_mut() {
-            *s = (*s)*invacc;
-        }
-        samples
     }
 }
 
 #[cfg(test)]
 mod test {
     use super::*;
 
+    //
+    // Check that the means of the components of n samples from
+    // the Dirichlet distribution agree with the expected means
+    // with a relative tolerance of rtol.
+    //
+    // This is a crude statistical test, but it will catch egregious
+    // mistakes.  It will also also fail if any samples contain nan.
+    //
```

Comment:
```suggestion
    /// Check that the means of the components of `n` samples from
    /// the Dirichlet distribution agree with the expected means
    /// with a relative tolerance of `rtol`.
    ///
    /// This is a crude statistical test, but it will catch egregious
    /// mistakes.  It will also also fail if any samples contain `nan`.
```

Reply:
Done.